Flaky test: `TestBuiltinVerifyHTTPPGPSignatureImpl`

https://github.com/docker/buildx/actions/runs/23822945008/job/69439356507#step:7:13967

```
=== Failed
=== FAIL: policy TestBuiltinVerifyHTTPPGPSignatureImpl/verify-failure-returns-false (0.00s)
    funcs_http_pgp_test.go:95: 
        	Error Trace:	/Users/runner/work/buildx/buildx/policy/funcs_http_pgp_test.go:95
        	Error:      	Not equal: 
        	            	expected: &ast.Term{Value:false, Location:(*location.Location)(nil)}
        	            	actual  : &ast.Term{Value:true, Location:(*location.Location)(nil)}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,3 +1,3 @@
        	            	 (*ast.Term)({
        	            	- Value: (ast.Boolean) false,
        	            	+ Value: (ast.Boolean) true,
        	            	  Location: (*location.Location)(<nil>)
        	Test:       	TestBuiltinVerifyHTTPPGPSignatureImpl/verify-failure-returns-false
    --- FAIL: TestBuiltinVerifyHTTPPGPSignatureImpl/verify-failure-returns-false (0.00s)

=== FAIL: policy TestBuiltinVerifyHTTPPGPSignatureImpl (0.16s)
```

--- a/policy/funcs_http_pgp_test.go
+++ b/policy/funcs_http_pgp_test.go
@@ -67,7 +67,11 @@ func TestBuiltinVerifyHTTPPGPSignatureImpl(t *testing.T) {
 	t.Run("verify-failure-returns-false", func(t *testing.T) {
 		encoded := checksumDigest.Encoded()
 		require.NotEmpty(t, encoded)
-		flipped := "0" + encoded[1:]
+		flippedFirst := byte('0')
+		if encoded[0] == '0' {
+			flippedFirst = '1'
+		}
+		flipped := string(flippedFirst) + encoded[1:]
 		badDigest := digest.NewDigestFromEncoded(checksumDigest.Algorithm(), flipped)
 
 		st := &state{

